### ISSUE TITLE
Having 2 functions with the same name should throw an error.

### ISSUE THREAD

**Issue by Remi-Gau** [florianschanda/miss_hit#235]
**What kind of feature is this?**

New feature in MISS_HIT

**Your MATLAB/Octave environment**

MATLAB

Octave is not affected by this

**MISS_HIT component affected**

I think this falls under the linter.

**Describe the solution you'd like**

Matlab will not use a file where 2 functions have the same name. I suspect that ms lint should flag this, no?

Example:

```
function temp()
end

function foo()
end

function foo()
end
```

**Comment by florianschanda** [comment]
Yes, it definitely should; but right now there is zero semantic analysis going on.

There is a sanity check that a function name doesn't clash with some important built-ins (like true, and yes this happened for real once, just remember all safety rules are written in blood) but other than that there is nothing going on: no symbol table no name resolution, nothing.

I have a very big project ahead of me to do this, as it would enable a large number of useful checks (e.g. look at [issues requiring sem](https://github.com/florianschanda/miss_hit/issues?q=is%3Aissue+is%3Aopen+label%3A%22requires%3A+sem%22) that are open right now).

I plan to do this; but don't expect anything this year...

**Comment by florianschanda** [comment]
In fact I've re-opened #236 and put the same labels on it to make sure I capture this

**Comment by Remi-Gau** [comment]
> I plan to do this; but don't expect anything this year...

No rush at all. 

miss_hit as it is has already brought much joy and removed a lot of my pain from my every day life. So anything beyond that is just icing on the cake. :+1:

**Comment by florianschanda** [comment]
That is some really nice feedback, it makes me very happy to hear that. Thanks! :)
